libvast/src/segment_store.cpp

```diff
@@ -373,8 +372,15 @@ void segment_store::inspect_status(caf::settings& xs,
 }
 
 caf::error segment_store::register_segments() {
-  for (auto filename : directory{segment_path()})
-    if (auto err = register_segment(filename))
+  std::error_code err{};
+  std::filesystem::directory_iterator dir{
+    std::filesystem::path{segment_path().str()}, err};
+  if (err)
```

Comment:
I just realized after reviewing in the first round: a non-existent directory is not a failure. In fact, if there is no persistent data yet, it's expected. So we may need something along the lines of `if (err != dir_does_not_exist)`.

Reply:
I now check at the beginning of the function if the segment directory exists before proceeding. Alternatively, I could perhaps check the `err` value with an error code I construct using a `generic_category` enum with a value of `2` for directory not existing, but I think this is clearer, though slightly less performant.  Note this is the value that `std::filesystem` maps for POSIX return codes to its error category/values for a directory not existing.

What do you think?

Reply:
> [..]  but I think this is clearer, though slightly less performant

Agreed; this is not on the critical path and in this case we benefit from clarity (and maybe even portability).